lib/smart_answer/calculators/holiday_entitlement.rb

```diff
@@ -106,13 +107,20 @@ def date_calc
       end
     end
 
-    def format_number(number, dp = 1)
-      str = sprintf("%.#{dp}f", number)
+    def format_number(number, decimal_places = 1)
+      scale = decimal_places.zero? ? 1.0 : decimal_places * 10.0
```

Comment:
This is wrong, it should be:
```suggestion
      scale = decimal_places ** 10
```

Digit-shift left by `decimal_places`, floor and shift right to get the rounded part, subtract to get discarded part, add `1 / scale` to rounded part if discarded part is nonzero.

Reply:
I'll add a test which catches this tomorrow.  I'm a little surprised the few tests which do 2dp worked, but I bet something using 3dp would not.